storage/store_metrics.go

```diff
@@ -141,6 +228,34 @@ type StoreMetrics struct {
 	RaftWorkingDurationNanos *metric.Counter
 	RaftTickingDurationNanos *metric.Counter
 
+	// Raft message metrics.
+	RaftRcvdMsgHup            *metric.Counter
+	RaftRcvdMsgBeat           *metric.Counter
+	RaftRcvdMsgProp           *metric.Counter
+	RaftRcvdMsgApp            *metric.Counter
+	RaftRcvdMsgAppResp        *metric.Counter
+	RaftRcvdMsgVote           *metric.Counter
+	RaftRcvdMsgVoteResp       *metric.Counter
+	RaftRcvdMsgSnap           *metric.Counter
+	RaftRcvdMsgHeartbeat      *metric.Counter
+	RaftRcvdMsgHeartbeatResp  *metric.Counter
+	RaftRcvdMsgUnreachable    *metric.Counter
+	RaftRcvdMsgSnapStatus     *metric.Counter
+	RaftRcvdMsgCheckQuorum    *metric.Counter
+	RaftRcvdMsgTransferLeader *metric.Counter
+	RaftRcvdMsgTimeoutNow     *metric.Counter
+
+	// A map for conveniently finding the appropriate metric. The individual
+	// metric references must exist as AddMetricStruct adds them by reflection
+	// on this struct and does not process map types.
+	// TODO(arjun): eliminate this duplication.
+	RaftRcvdMessages       map[raftpb.MessageType]*metric.Counter
```

Comment:
let's at least make it private (s/RaftRcvdMessages/raftRcvdMessage/). `AddMetricStruct` will ignore it due to be inaccessible which is a bit clearer than being the wrong type.